src/nl/surf/eduhub_rio_mapper/metrics.clj

```diff
@@ -1,12 +1,43 @@
 (ns nl.surf.eduhub-rio-mapper.metrics
-  (:require [clojure.string :as str]))
+  (:require [clojure.string :as str]
+            [nl.surf.eduhub-rio-mapper.worker :as worker]))
 
-(defn render-metrics [queue-count]
-  {:pre [(map? queue-count)
-         (every? string? (keys queue-count))
-         (every? integer? (vals queue-count))]}
-  (str/join "\n" (map (fn [[k v]] (format "rio_mapper_active_and_queued_job_count{schac_home=\"%s\"} %s" k v))
-                      queue-count)))
+;; Retrieves the total number of processed jobs by status (started,done,error,time_out)
+;; The job count is grouped per schachome. Returns a map with as keys the status (keyword)
+;; and as values vectors of vectors with schachome (string) and count (integer).
+(defn fetch-jobs-by-status-count [config]
+  (let [process-pair
+        (fn [[k cnt]]
+          (let [[schach-home status] (clojure.string/split k #"/")]
+            [status schach-home cnt]))
+        process-triplet
+        (fn [h [status schach-home cnt]]
+          (assoc-in h [status schach-home] cnt))]
+    (->> (worker/fetch-hash config worker/jobs-count-by-status-key-name)
+         (partition 2)
```

Comment:
Dit is wel een indicatie dat dit niet logisch in elkaar zit:

 - worker/fetch-hash geeft dus blijkbaar geen hash terug.
 - worker/jobs-count-by-status-key-name moet je doorgeven maar dat is een constant en de enige waarde die ooit wordt doorgegeven.
 - hier doe je een string/split waarvan je dan weet hoe die hash key in elkaar zit maar dat is gedefinieerd in de worker namespace.

Conclusie: maak een functie in worker die die metrics teruggeeft in een format dat direct bruikbaar is.

Reply:
worker/fetch-hash geeft de data van een redis hash terug, dus een generieke wrapper om HGETALL.
worker/increment-hash-key is een generieke wrapper om HINCRBY.
Ik wil metrics-gerelateerde functies in metrics, niet in worker.